### ISSUE TITLE
"matches" operator in URL condition breaks responce body

### ISSUE THREAD

**Issue by drudoi** [mcustiel/phiremock#19]
Using "matches" operator for URL I have unexpecred response.

For this expectation:
```JSON
{
  "request": {
    "method": "GET",
    "url": {
      "matches": "~/hello~"
    }
  },
  "response": {
    "statusCode": 200,
    "body": "Hello!"
  }
}
```
I have the next result:
```
$ curl http://localhost:8086/hello
http://localhost:8086Hello!
```

And if the condition is strict (otherwise it seen on any match, for `/i/dont/want/hello` too)
```JSON
{
  "request": {
    "method": "GET",
    "url": {
      "matches": "~^/hello$~"
    }
  },
  "response": {
    "statusCode": 200,
    "body": "Hello!"
  }
}
```
I see the next:
```
$ curl http://localhost:8086/hello
http://localhost:8086/hello
```

It happens because of replacement in `\Mcustiel\Phiremock\Server\Utils\Strategies\RegexResponseStrategy`:
```php
    private function fillWithUrlMatches($expectation, $httpRequest, $responseBody)
    {
        if ($this->urlConditionIsRegex($expectation)) {
            $responseBody = preg_replace('/\$\{url\.(\d+)\}/', '\$$1', $responseBody);

            return preg_replace(
                $expectation->getRequest()->getUrl()->getValue(),
                $responseBody,
                $httpRequest->getUri()->__toString()
            );
        }

        return $responseBody;
    }
```

Here `$expectation->getRequest()->getUrl()->getValue()` is `"~/hello~"`, `$responseBody` is `"Hello!"`, and `$httpRequest->getUri()->__toString()` is `"http://localhost:8086/hello"`. The last value adds missmatch with search argument in response, and outputs if the condition isn't matches.

I don't know how to fix it in the best way, I'll think about it tommorow. I'll be happy to see ideas in comments.

**Comment by mcustiel** [comment]
Wow, nice catch. Thank you for reporting it.

**Comment by mcustiel** [comment]
Released in v1.6.0. Thank you @dewa1t
